coderd/templateversions_test.go

```diff
@@ -125,6 +125,12 @@ func TestPatchCancelTemplateVersion(t *testing.T) {
 		var apiErr *codersdk.Error
 		require.ErrorAs(t, err, &apiErr)
 		require.Equal(t, http.StatusPreconditionFailed, apiErr.StatusCode())
+		require.Eventually(t, func() bool {
+			var err error
+			version, err = client.TemplateVersion(context.Background(), version.ID)
+			require.NoError(t, err)
```

Comment:
This is a thing that I like to keep a watch out for as it burned me a couple times in the past with flaky tests -- `require.Eventually` runs its callback in a separate goroutine to the testmain goroutine. Any function from `github.com/stretchr/testify/require` can end up calling `t.FailNow` which is unsafe to do in anything besides the testmain. However each `require` has a corresponding `assert` that doesn't have this problem.

https://pkg.go.dev/testing#T.FailNow

As a bonus, the `assert.*` versions can also be chained as booleans, which lets you write neat stuff like this:

```suggestion
			return assert.NoError(t, err) &&
			    version.Job.Status == codersdk.ProvisionerJobFailed
```

Reply:
I can attest to this being sucky! I believe @johnstcn fixed up all the times I messed this up, and I didn't know about it until he pointed it out! 🙈

Reply:
Ahh, good to know. Thanks you for the explanation and snippet.

Reply:
Seconded (or is that thirded)! I’ve also noticed there are other dangers to using `require` because of it’s use of `FailNow`. For example, it’s easy to miss that `require` is essentially `if err != nil { return }`, so assumptions like a connection will be closed further down the line break, and it diverges from the mental model of regular Go code.